lib/matplotlib/backends/_backend_tk.py

```diff
@@ -146,7 +150,14 @@ def _on_timer(self):
         # don't recreate the timer in that case.
         if not self._single and self._timer:
             if self._interval > 0:
-                self._timer = self.parent.after(self._interval, self._on_timer)
+                # We want to adjust our fire time independent of the time
+                # spent in the callback and not drift over time.
+                elapsed_ms = int((time.perf_counter_ns() - self._timer_orig_fire_time)
```

Comment:
The int() shouldn't be necessary here. Are you causing a cast to float by using the 1e6 literal?

Reply:
Yes, good call on that, I've changed it to `// 1_000_000` locally now. I've got other issues on the macos runners it looks like and I think some of them may be due to fixes that are in another backend timers PR: https://github.com/matplotlib/matplotlib/pull/29018
I'll hold off on pushing now until I get a bit more investigation into the other failures to save some CI runs.